src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+                check_e_bins(
+                    bins=self.ang_res.reco_energy_bins,
+                    source="Angular resolution energy",
+                )
+            self.bias_res = EnergyBiasResolutionMakerBase.from_name(
+                self.energy_bias_res_parameterization, parent=self
+            )
+            self.sens = SensitivityMakerBase.from_name(
+                self.sens_parameterization, parent=self
+            )
+            check_e_bins(
+                bins=self.sens.reco_energy_bins, source="Sensitivity reco energy"
+            )
+
+    def calculate_selections(self, reduced_events: dict) -> dict:
+        """
+        Add the selection columns to the signal and optionally background tables.
+
+        Parameters
+        ----------
+        reduced_events: dict
+            dict containing the signal (``"gammas"``) and optionally background
+            tables (``"protons"``, ``"electrons"``)
+
+        Returns
+        -------
+        dict
+            ``reduced_events`` with selection columns added.
+        """
+        reduced_events["gammas"]["selected_gh"] = evaluate_binned_cut(
+            reduced_events["gammas"]["gh_score"],
+            reduced_events["gammas"]["reco_energy"],
+            self.opt_result.gh_cuts,
+            operator.ge,
+        )
+        if not self.full_enclosure:
+            reduced_events["gammas"]["selected_theta"] = evaluate_binned_cut(
+                reduced_events["gammas"]["theta"],
+                reduced_events["gammas"]["reco_energy"],
+                self.opt_result.theta_cuts,
+                operator.le,
+            )
+            reduced_events["gammas"]["selected"] = (
+                reduced_events["gammas"]["selected_theta"]
+                & reduced_events["gammas"]["selected_gh"]
+            )
+        else:
+            reduced_events["gammas"]["selected"] = reduced_events["gammas"][
+                "selected_gh"
+            ]
+
+        if self.do_background:
+            for bg_type in ("protons", "electrons"):
+                reduced_events[bg_type]["selected_gh"] = evaluate_binned_cut(
+                    reduced_events[bg_type]["gh_score"],
+                    reduced_events[bg_type]["reco_energy"],
+                    self.opt_result.gh_cuts,
+                    operator.ge,
+                )
+
+        if self.do_background:
+            self.log.debug(
+                "Keeping %d signal, %d proton events, and %d electron events"
+                % (
+                    sum(reduced_events["gammas"]["selected"]),
+                    sum(reduced_events["protons"]["selected_gh"]),
+                    sum(reduced_events["electrons"]["selected_gh"]),
+                )
+            )
+        else:
+            self.log.debug(
+                "Keeping %d signal events" % (sum(reduced_events["gammas"]["selected"]))
+            )
+        return reduced_events
+
+    def _stack_background(self, reduced_events):
```

Comment:
this function seems needlessly complex.

How about:

```
if self.proton_file and self.electron_file:
    return vstack([reduced_events["protons"], reduced_events["electrons"])

if self.proton_file:
    return reduced_events["proton"]

return reduced_events["electron"]
```

but the same comment as above also applies here: i think we shouldn't support only running with electrons as background.

Reply:
When requiring the proton file, this turned into only 4 lines, so I removed this helper function and did it inline instead.